src/daft-connect/src/spark_analyzer.rs

```diff
@@ -294,22 +293,117 @@ impl SparkAnalyzer<'_> {
             debug!("Ignoring options: {options:?}; not yet implemented");
         }
 
-        if !predicates.is_empty() {
-            debug!("Ignoring predicates: {predicates:?}; not yet implemented");
+        // TODO: allow user to set handling for unused options. either ignore or error.
+        fn check_unused_options(
+            format: &str,
+            options: &HashMap<String, String>,
+        ) -> ConnectResult<()> {
+            let mut unimplemented_options = Vec::new();
+            for (key, value) in options {
+                warn!("Option '{key}' is not yet implemented");
+                unimplemented_options.push(format!(r#""{}"={}"#, key, value));
+            }
+            if unimplemented_options.is_empty() {
+                Ok(())
+            } else {
+                Err(ConnectError::not_yet_implemented(format!(
+                    "[{unimplemented_options}] options for {format}",
+                    unimplemented_options = unimplemented_options.join(", ")
+                )))
+            }
         }
 
-        Ok(match &*format {
-            "parquet" => ParquetScanBuilder::new(paths).finish().await?,
-            "csv" => CsvScanBuilder::new(paths).finish().await?,
+        let format = &*format;
+        Ok(match format {
+            "parquet" => {
+                let chunk_size = options.remove("chunk_size").and_then(|v| v.parse().ok());
+                let hive_partitioning = options
+                    .remove("hive_partitioning")
+                    .and_then(|v| v.parse().ok());
+                let mut builder = ParquetScanBuilder::new(paths);
+                builder.chunk_size = chunk_size;
+
+                if let Some(hive_partitioning) = hive_partitioning {
+                    builder.hive_partitioning = hive_partitioning;
+                }
+                check_unused_options(format, &options)?;
+
+                builder.finish().await?
+            }
+            "csv" => {
+                // reference for csv options:
+                // https://spark.apache.org/docs/latest/sql-data-sources-csv.html
+                let mut builder = CsvScanBuilder::new(paths);
+
+                if let Some(sep) = options.remove("sep").and_then(|s| s.chars().next()) {
+                    builder = builder.delimiter(sep);
+                }
+
+                // spark sets this to false by default, so we'll do the same
+                let header = options
+                    .remove("header")
+                    .and_then(|v| v.to_lowercase().parse().ok())
+                    .unwrap_or(false);
```

Comment:
Would this make `header=1` return false? Let's validate the input and err if not a bool. I don't necessarily think we should support header=1, but just want the same end behavior as connect. 

Considering daft already uses =1 for true with its environment variables, this could be perceived as a bug. For now, let's validate input.

Reply:
`1` does not parse to a boolean. here's a quick demo. 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=160892e1e3f90c291d17cd09788f0137

Reply:
Correct, my comment is about erroring on input validation because this silently returns a default when input is invalid.

Reply:
```rust
let b: bool = "1".parse().ok().unwrap_or(false); // <- unwrap hides err
assert!(b == false) // should have err'd!
```

To be more explicit here .. **I do not think we should support 1->true  for daft-connect** but should validate the input and error if the value is not supported by spark's options spec.

As-is this could be interpreted as a bug or footgun since 1<->true in other parts of daft.